pkg/metricscollector/v1beta1/file-metricscollector/file-metricscollector_test.go

```diff
@@ -215,3 +352,57 @@ func generateTestFiles() error {
 
 	return nil
 }
+
+func generateTEXTTestFiles() error {
+	if _, err := os.Stat(testTextDataPath); err != nil {
+		if err = os.MkdirAll(testTextDataPath, 0700); err != nil {
+			return err
+		}
+	}
```

Comment:
Should we move directory creation outside of these functions and create it before we [invoke them](https://github.com/kubeflow/katib/blob/42f2d458e5437210ac77febb24cbc07fab28d761/pkg/metricscollector/v1beta1/file-metricscollector/file-metricscollector_test.go#L37) ?

Reply:
Both are okay. But keeping them in `generateXXXTestFiles` will make the code look more tidy.

Reply:
That's true, but since we always run these functions sequentially why do we need to create directories in each of them ?
Maybe we could have another function called: `generateTestDirectory` which will create `testdata` dir.
WDYT @Electronic-Waste @tenzen-y ?

Reply:
@andreyvelich That sounds reasonable. Having separate function wouldn't be wroth it.

Reply:
I meant that consolidating 2 generators into 1 would be better.

Reply:
Okay, that sounds reasonable. I'll modify it soon.

Reply:
@andreyvelich @tenzen-y I also integrate `os.RemoveAll` operations into the func `deleteTestDirs()`. plz cc👀